packages/mask/src/plugins/Referral/SNSAdaptor/AdjustFarmRewards.tsx

```diff
@@ -0,0 +1,379 @@
+    useChainId,
+    useFungibleTokenBalance,
+    useWeb3,
+    FungibleTokenDetailed,
+} from '@masknet/web3-shared-evm'
+
+import { AdjustFarmRewardsInterface, TransactionStatus, PagesType, TabsReferralFarms } from '../types'
+import { useI18N } from '../../../utils'
+import { EthereumChainBoundary } from '../../../web3/UI/EthereumChainBoundary'
+import ActionButton from '../../../extension/options-page/DashboardComponents/ActionButton'
+import { useRequiredChainId } from '../hooks/useRequiredChainId'
+import { roundValue } from '../helpers'
+import { APR, ATTRACE_FEE_PERCENT } from '../constants'
+import { referralFarmService, farmsService } from '../Worker/services'
+
+import { FarmTokenDetailed } from './shared-ui/FarmTokenDetailed'
+
+import { useSharedStyles } from './styles'
+
+const useStyles = makeStyles()((theme) => ({
+    valueCol: {
+        display: 'flex',
+        flexDirection: 'column',
+        '& span': { fontWeight: 600, marginTop: '4px' },
+    },
+    balance: {
+        whiteSpace: 'nowrap',
+        textOverflow: 'ellipsis',
+        overflow: 'hidden',
+        maxWidth: '80%',
+        fontSize: 12,
+        top: theme.spacing(0.5),
+    },
+    textField: {
+        '& input[type=number]': {
+            height: 30,
+            '-moz-appearance': 'textfield',
+        },
+        '& input[type=number]::-webkit-outer-spin-button': {
+            '-webkit-appearance': 'none',
+            margin: 0,
+        },
+        '& input[type=number]::-webkit-inner-spin-button': {
+            '-webkit-appearance': 'none',
+            margin: 0,
+        },
+    },
+}))
+
+export function AdjustFarmRewards(props: AdjustFarmRewardsInterface) {
+    const { farm, rewardToken, referredToken } = props
+    localStorage.setItem('adjustFarmRewardsData', JSON.stringify({ farm, rewardToken, referredToken }))
+
+    const { t } = useI18N()
+    const { classes } = useStyles()
+    const { classes: sharedClasses } = useSharedStyles()
+    const chainId = useChainId()
+    const web3 = useWeb3({ chainId })
+    const account = useAccount()
+    const { showSnackbar } = useCustomSnackbar()
+    const requiredChainId = useRequiredChainId(chainId)
+    const { value: rewardBalance = '0' } = useFungibleTokenBalance(
+        rewardToken?.type ?? EthereumTokenType.Native,
+        rewardToken?.address ?? '',
+    )
+
+    const [attraceFee, setAttraceFee] = useState<number>(0)
+    const [dailyFarmReward, setDailyFarmReward] = useState<string>('')
+    const [totalFarmReward, setTotalFarmReward] = useState<string>('')
+
+    const { value: farmsMetaState } = useAsync(
+        async () => (farm?.farmHash ? farmsService.getFarmsMetaState(chainId, [farm.farmHash]) : undefined),
+        [farm, chainId],
+    )
+
+    const onChangeTotalFarmReward = useCallback((e: React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement>) => {
+        const totalFarmReward = e.currentTarget.value
+        const totalFarmRewardNum = Number(totalFarmReward)
```

Comment:
Recommend to use `Number.parseInt` or `Number.parseFloat`.

Reply:
plz enable https://dimensiondev.github.io/eslint-plugin/src/rules/type/no-number-constructor